spec/services/participants/change_schedule/early_career_teacher_spec.rb

```diff
@@ -3,4 +3,150 @@
+      let!(:induction_record) do
+        create(
+          :induction_record,
+          participant_profile: profile,
+          schedule: original_schedule,
+          induction_programme: induction_programme,
+        )
+      end
+      let(:induction_programme) do
+        create(
+          :induction_programme,
+          :fip,
+          school_cohort: school_cohort,
+          partnership: partnership,
+        )
+      end
+
+      subject do
+        described_class.new(params: {
+          schedule_identifier: schedule.schedule_identifier,
+          participant_id: user.id,
+          course_identifier: "ecf-induction",
+          cpd_lead_provider: cpd_lead_provider,
+        })
+      end
+
+      it "should not have an error" do
+        expect { subject.call }.not_to raise_error
+      end
+
+      it "updates the profile#schedule" do
+        expect { subject.call }.to change { profile.reload.schedule }.from(original_schedule).to(schedule)
+      end
+
+      it "updates induction_record#schedule" do
+        expect { subject.call }.to change { induction_record.reload.schedule }.from(original_schedule).to(schedule)
+      end
+    end
+
+    context "when changing to schedule not suitable for the course" do
+      let(:cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider) }
+      let(:lead_provider) { cpd_lead_provider.lead_provider }
+      let(:user) { profile.user }
+      let(:profile) { create(:ect_participant_profile, school_cohort: school_cohort) }
+      let(:school_cohort) { create(:school_cohort) }
+      let(:school) { school_cohort.school }
+      let!(:partnership) { create(:partnership, school: school, lead_provider: lead_provider, cohort: school_cohort.cohort) }
+      let(:schedule) { create(:ecf_mentor_schedule) }
+
+      subject do
+        described_class.new(params: {
+          schedule_identifier: schedule.schedule_identifier,
+          participant_id: user.id,
+          course_identifier: "ecf-induction",
+          cpd_lead_provider: cpd_lead_provider,
+        })
+      end
+
+      it "should have an error" do
+        expect { subject.call }.to raise_error(ActionController::ParameterMissing)
+      end
+    end
+  end
+
+  describe "changing to a soft schedules with previous declarations" do
+    let(:cpd_lead_provider) { create(:cpd_lead_provider, :with_lead_provider) }
+    let(:lead_provider) { cpd_lead_provider.lead_provider }
+    let(:user) { profile.user }
+    let(:profile) { create(:ect_participant_profile, school_cohort: school_cohort) }
+    let(:school_cohort) { create(:school_cohort) }
+    let(:cohort) { school_cohort.cohort }
+    let(:school) { school_cohort.school }
+    let!(:partnership) { create(:partnership, school: school, lead_provider: lead_provider, cohort: school_cohort.cohort) }
+    let(:schedule) { create(:ecf_mentor_schedule) }
+
+    let(:schedule) do
+      Finance::Schedule.create!(
```

Comment:
Suggestion: use `FactoryBot` to be consistent with the rest of the code

Reply:
done